control/metrics.go

```diff
@@ -289,7 +301,7 @@ func (mc *metricCatalog) AddLoadedMetricType(lp *loadedPlugin, mt core.Metric) e
 		return err
 	}
 	newMt := metricType{
-		Plugin:             lp,
+		Plugin:             *lp,
```

Comment:
Consider creating a `catalogedPlugin` struct and `newCatalogedPlugin` function that returns a core.CatalogedPlugin (implementing a deep copy).